### ISSUE TITLE
game quits with "Cause: Access Violation  11 Data Execution Prevention (DEP) at: 0x00000000^M"

### ISSUE THREAD

**Issue by Jeremiah-Roise** [R2Northstar/Northstar#147]
### Bug
Expected: game starts
what actually happens origin launches blank Titanfall2 window appears console appears crash notification pops up
![image](https://user-images.githubusercontent.com/56651804/149064103-262bfa21-2e56-4864-87eb-6231f7e58e08.png)
![image](https://user-images.githubusercontent.com/56651804/149064165-16f2fa19-691b-439b-890b-bfc024dc7e71.png)

### Steps to reproduce

  1. install northstar in steam
  2. rename vanilla Titanfall2 to Titanfall2.exe.old
  3. name Northstarlauncher.exe to Titanfall2.exe
  4. start game from steam with proton version 6.3-8, proton experimental, or 5.13-6

### Specifications
  - Northstar version: v1.4.0
  - Platform: Steam, Manjaro, i3wm-gaps,
   here is my neofetch:
OS: Manjaro Linux x86_64
Host: GF65 Thin 9SEXR REV:1.0
Kernel: 5.10.89-1-MANJARO
Uptime: 3 hours, 21 mins
Packages: 1449 (pacman), 13 (flatpak)
Shell: bash 5.1.12
Resolution: 1920x1080, 1920x1080
DE: Xfce 4.16
WM: i3
Theme: Matcha-dark-sea [GTK2/3]
Icons: Papirus-Maia [GTK2], Adwaita [GTK3]
Terminal: alacritty
CPU: Intel i7-9750H (12) @ 4.500GHz
GPU: NVIDIA GeForce RTX 2060 Mobile
GPU: Intel CoffeeLake-H GT2 [UHD Graphics 630]
Memory: 4119MiB / 15846MiB
[nslog2022-01-11 20-45-59.txt](https://github.com/R2Northstar/Northstar/files/7851683/nslog2022-01-11.20-45-59.txt)

**Comment by Clos3y** [comment]
I'm getting the same error. I've tried disabling DEP running commands via `/tmp/proton_${USER}/run`, but the commands necessary don't exist.

[nslog2022-01-13 14-55-51.txt](https://github.com/R2Northstar/Northstar/files/7863697/nslog2022-01-13.14-55-51.txt)

**Comment by GeckoEidechse** [comment]
What happens if you keep the files named back to the original way, i.e. `Titanfall2.exe.old` -> `Titanfall2.exe` ..., add `-northstar` to the launch arguments for Titanfall 2 in Steam and then launch Titanfall 2 via Steam?
(This method is commonly used when Origin causes problems)

Before `v1.4.0` Northstar actually used the `unpacked.exe` that is included in the zip instead of `Titanfall2.exe`, with the launcher rewrite in `v1.4.0` the `Titanfall2.exe` is hooked directly. I'm assuming the rename causes issues in that regard as the launcher will then try to run itself.

**Comment by Clos3y** [comment]
> What happens if you keep the files named back to the original way, i.e. `Titanfall2.exe.old` -> `Titanfall2.exe` ..., add `-northstar` to the launch arguments for Titanfall 2 in Steam and then launch Titanfall 2 via Steam?

Didn't fix it for me. Unzipped 1.4.0 fresh into the directory and added the argument, but it just launches normal Titanfall 2. Although my Titanfall 2 is now borked, might have to reinstall.

**Comment by ChunkyPanda03** [comment]
when it does 

> EDIT: Yeah... my Titanfall install is now completely borked. It keeps running OriginThinSetup.

clear the prefix however I have the same crashing issue and I have also been able to run the normal Titanfall 2 on my linux install so I am not sure why it is giving me the DEP issue

as a side not could it be that we are using NVIDIA graphic cards?

**Comment by LiamDawe** [comment]
The main game works just fine on here with NVIDIA + Linux, it's just Northstar that's not working for me for the exact issue described.

**Comment by Clos3y** [comment]
> clear the prefix 

I did, but it's mega borked. Although now I'm getting other issues after a system update.

**Comment by Jeremiah-Roise** [comment]
> What happens if you keep the files named back to the original way, i.e. `Titanfall2.exe.old` -> `Titanfall2.exe` ..., add `-northstar` to the launch arguments for Titanfall 2 in Steam and then launch Titanfall 2 via Steam? (This method is commonly used when Origin causes problems)
> 

this didn't work for me but I see what your saying.

**Comment by skyegallup** [comment]
I am running across the same issue with Titanfall 2 installed via Steam when using the official installation instructions (unpack archive into game folder, rename NorthstarClient.exe to Titanfall2.exe, run via Steam launcher).

System info: Pop!_OS 21.04, Proton Experimental, Northstar 1.4.0, Ryzen 5 2600X, GTX 1050 Ti (with somewhat up-to-date graphics drivers)

[nslog2022-01-14 21-39-58.txt](https://github.com/R2Northstar/Northstar/files/7875343/nslog2022-01-14.21-39-58.txt)

> What happens if you keep the files named back to the original way, i.e. `Titanfall2.exe.old` -> `Titanfall2.exe` ..., add `-northstar` to the launch arguments for Titanfall 2 in Steam and then launch Titanfall 2 via Steam?

This starts the vanilla game, though I get a 429 error when attempting to connect to matchmaking servers. I'm not sure if the error is because I have Northstar files with a vanilla client or because the official matchmaking servers are overloaded.

**Comment by GeckoEidechse** [comment]
> This starts the vanilla game, though I get a 429 error when attempting to connect to matchmaking servers. I'm not sure if the error is because I have Northstar files with a vanilla client or because the official matchmaking servers are overloaded.

429 error is solely vanilla related. I assumed `-northstar` also works on Linux but judging by the responses this doesn't seem to be the case.

**Comment by RoamingNoMaD** [comment]
Also had the same problem, running it with Lutris+wine. 

OS: Manjaro Linux x86_64 
Kernel: 5.10.89-1-MANJARO 
Resolution: 1920x1080 
DE: Plasma 5.23.4 
CPU: Intel i5-10300H (8) @ 4.500GHz 
GPU: NVIDIA GeForce GTX 1650 Ti Mobile 
GPU: Intel CometLake-H GT2 [UHD Graphics] 

Bonus crashlog:

[nslog2022-01-18 00-51-25.txt](https://github.com/R2Northstar/Northstar/files/7885214/nslog2022-01-18.00-51-25.txt)

**Comment by schrodinger44** [comment]
as of today, i have managed to run northstar using the newest version of glorious eggroll

**Comment by RoamingNoMaD** [comment]
You're right, i managed to get the game working by making a link to the Proton experimental folders into Lutris, and using it as a Wine runner. I think this solves this issue for people who have the game on Origin.

**Comment by abarichello** [comment]
This fix is now part of our wiki, check this link if you are having the same problem

**Comment by GeckoEidechse** [comment]
In particular `Proton-7.3-GE-1` was found to resolve this issue.

Wiki entry can be [found here](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux#game-crashes-on-launch-with-cause-access-violation-data-execution-prevention-dep-at-0x00000000).

**Comment by GeckoEidechse** [comment]
Log and dump when running Northstar on my SteamDeck according to our Linux instructions with `Proton 7.0-1`:
[nslog2022-03-17 21-55-32.txt](https://github.com/R2Northstar/Northstar/files/8292562/nslog2022-03-17.21-55-32.txt)
[nsdump2022-03-17 21-55-32.zip](https://github.com/R2Northstar/Northstar/files/8292563/nsdump2022-03-17.21-55-32.zip)

**Comment by GeckoEidechse** [comment]
We've since updated the wiki with instructions for the specific Proton version to use. Wiki entry can be [found here](https://r2northstar.gitbook.io/r2northstar-wiki/using-northstar/playing-on-linux#game-crashes-on-launch-with-cause-access-violation-data-execution-prevention-dep-at-0x00000000).

@Jeremiah-Roise could you give this a shot?

General compatibility improvements are discussed in #214

**Comment by GeckoEidechse** [comment]
Closing due to lack of response under the assumption that linked wiki entry fixes issue. If not please re-open or make a new issue referencing this one.

**Comment by nickdibart16** [comment]
I am having this issue but I am running Windows 10 pro. The provided links don't work (or no longer work).

**Comment by GeckoEidechse** [comment]
> I am having this issue but I am running Windows 10 pro. The provided links don't work (or no longer work).

I updated the link. This is in relation to Linux though. @nickdibart16 please [open a new GitHub issue](https://github.com/R2Northstar/Northstar/issues/new) describing your set up and the steps to reproduce the crash (if reproducible).
